src/dhcp_proxy/dhcp_service.rs

```diff
@@ -49,6 +49,11 @@ impl DhcpV4Service {
     pub fn new(nc: NetworkConfig, timeout: u32) -> Result<Self, DhcpServiceError> {
         let mut config = DhcpV4Config::new_proxy(&nc.host_iface, &nc.container_mac_addr);
         config.set_timeout(timeout);
+
+        if !nc.host_name.is_empty() {
+            config.set_host_name(&nc.host_name);
+            config.use_host_name_as_client_id();
```

Comment:
Is there a reason to set the host name as client id? I have not looked at other DHCP clients if they do that by default but I would assume just setting the host name alone should be good enough for server to consider the name?

Reply:
DHCP servers look for existing leases by client id which is normally derived from the MAC address.  By default though a random MAC address gets assigned every time you (re)start a container so if you stop and start a container in quick succession, you could get a different IP address. Because of DNS TTL, hosts could have the old address cached and possibly not be able to reach the container.  By using the hostname as the client id, you _should_ be able to restart a container and get the same IP address.

Reply:
Yes that makes sense thanks.

Reply:
@SilverBut pointed out a potential issue here... https://github.com/containers/netavark/issues/676#issuecomment-2501696991

I'm going to re-think a bit.

Reply:
I've made the client id the first 16 characters of the container ID which should resolve the concerns.  The way I did it however is a workaround because mozim doesn't provide a way to explicitly set the client id on DhcpV4Config.  I'm going to see if I can get that added so we don't need the workaround.